korman/exporter/animation.py

```diff
@@ -73,37 +73,55 @@ def fetch_animation_data(id_data):
         # to be listed on the SceneObject in a specific order. D'OH! So, always use these funcs.
         agmod, agmaster = self.get_anigraph_objects(bo, so)
         atcanim = self._mgr.find_create_object(plATCAnim, so=so)
+        globalanim = self._mgr.find_create_object(plAgeGlobalAnim, so=so)
+        anim_data = bo.plasma_modifiers.animation
 
-        # Add the animation data to the ATC
+        # Add the animation data to the ATC or, if there's an SDL variable, the Age Global Animation
         for i in applicators:
-            if i is not None:
+            if i is not None and anim_data.obj_sdl_anim is None:
                 atcanim.addApplicator(i)
+            elif i is not None and anim_data.obj_sdl_anim:
+                globalanim.addApplicator(i)                
         agmod.channelName = bo.name
-        agmaster.addPrivateAnim(atcanim.key)
+        if anim_data.obj_sdl_anim:
+            agmaster.addPrivateAnim(globalanim.key)
+        else:
+            agmaster.addPrivateAnim(atcanim.key)
 
         # This was previously part of the Animation Modifier, however, there can be lots of animations
         # Therefore we move it here.
         def get_ranges(*args, **kwargs):
             index = kwargs.get("index", 0)
+            anim_data = bo.plasma_modifiers.animation
             for i in args:
                 if i is not None:
                     yield i.frame_range[index]
-        atcanim.name = "(Entire Animation)"
-        atcanim.start = self._convert_frame_time(min(get_ranges(obj_action, data_action, index=0)))
-        atcanim.end = self._convert_frame_time(max(get_ranges(obj_action, data_action, index=1)))
+            if anim_data.obj_sdl_anim:
+                globalanim.name = "(Entire Animation)"
+                globalanim.start = self._convert_frame_time(min(get_ranges(obj_action, data_action, index=0)))
+                globalanim.end = self._convert_frame_time(max(get_ranges(obj_action, data_action, index=1)))            
+            else:
+                atcanim.name = "(Entire Animation)"
+                atcanim.start = self._convert_frame_time(min(get_ranges(obj_action, data_action, index=0)))
+                atcanim.end = self._convert_frame_time(max(get_ranges(obj_action, data_action, index=1)))
 
         # Marker points
         if obj_action is not None:
             for marker in obj_action.pose_markers:
                 atcanim.setMarker(marker.name, self._convert_frame_time(marker.frame))
 
+        def set_anim_params(plAGAnim):
+            sdl_name = bo.plasma_modifiers.animation.obj_sdl_anim
+            if sdl_name is not None:
+                globalanim.fGlobalVarName = sdl_name
```

Comment:
I think this should be `globalanim.globalVarName`

Reply:
Yep, that fix that part at least. Thank you!

Reply:
Actually, after redoing the script a bit, it's back to not working. ☹️